src/badge/index.tsx

```diff
@@ -12,17 +13,34 @@ import styles from './styles.css.js';
 
 export { BadgeProps };
 
-export default function Badge({ color = 'grey', children, ...rest }: BadgeProps) {
+export default function Badge({ color = 'grey', children, style, ...rest }: BadgeProps) {
   const { __internalRootRef } = useBaseComponent('Badge', { props: { color } });
   const baseProps = getBaseProps(rest);
 
   const className = clsx(baseProps.className, styles.badge, styles[`badge-color-${color}`]);
 
   return (
-    <span {...baseProps} {...{ className }} ref={__internalRootRef}>
+    <span {...baseProps} {...{ className }} ref={__internalRootRef} style={getBadgeStyles(style)}>
       {children}
     </span>
   );
 }
 
 applyDisplayName(Badge, 'Badge');
+
+function getBadgeStyles(style: BadgeProps['style']) {
```

Comment:
Let's move this into a separate style.tsx file and establish that as a pattern. Check out the Flashbar for a reference. I'm also not sure about this BadgeProps['style'] typing syntax.